internal/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -301,18 +315,22 @@ func (r *RegoPolicyInterpreter) EnableLogging(path string, level LogLevel) error
 	r.resultsLogger = log.New(file, "RESULT: ", log.Ldate|log.Ltime)
 	r.metadataLogger = log.New(file, "METADATA: ", log.Ldate|log.Ltime)
 	r.logInfo("Logging Enabled at level %d", level)
+
+	r.compiledModules = nil
```

Comment:
Are we flushing previously compiled modules, so that they have to be re-compiled using the new logging information?

Reply:
Also, should `dataAndModulesMutex` be held if we are resetting `compiledModules`?

Reply:
Yes, we have to flush the old compiled modules because the flag to enable printing has to be compiled into the module. The lock does need to be held, I've added it.